diff --git a/kdesu/kdesud/CMakeLists.txt b/kdesu/kdesud/CMakeLists.txt
index ca192bc..52e7636 100644
--- a/kdesu/kdesud/CMakeLists.txt
+++ b/kdesu/kdesud/CMakeLists.txt
@@ -3,7 +3,6 @@ check_function_exists(getpeereucred HAVE_GETPEERUCRED) # solaris style
 
 configure_file (config-kdesud.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kdesud.h )
 
-
 set(kdesud_SRCS
    kdesud.cpp
    repo.cpp
@@ -13,6 +12,8 @@ set(kdesud_SRCS
 
 kde4_add_executable(kdesud NOGUI ${kdesud_SRCS})
 
+SET_TARGET_PROPERTIES(kdesud PROPERTIES LINK_FLAGS "-Wl,-z,relro,-z,now")
+
 target_link_libraries(kdesud ${KDE4_KDESU_LIBS} ${KDE4_KDECORE_LIBS} ${X11_LIBRARIES})
 
 if(KDE4_ENABLE_FPIE)
